src/esp/sim/Simulator.cpp

```diff
@@ -117,171 +126,405 @@ void Simulator::reconfigure(const SimulatorConfiguration& cfg) {
+    const physics::MotionType attrObjMotionType =
+        static_cast<physics::MotionType>(objInst->getMotionType());
+    const physics::MotionType objMotionType =
+        physicsManager_->getObjectMotionType(objID);
+
+    if ((attrObjMotionType != objMotionType) &&
+        (attrObjMotionType != physics::MotionType::UNDEFINED)) {
+      physicsManager_->setObjectMotionType(objID, attrObjMotionType);
+    }
+    objectsAdded.push_back(objID);
+  }  // for each object attributes
+  // objectsAdded holds all ids of added objects.
 
-    const Magnum::Range3D& sceneBB = rootNode.computeCumulativeBB();
-    resourceManager_->setLightSetup(gfx::getDefaultLights());
-
-    // set activeSemanticSceneID_ values and push onto sceneID vector if
-    // appropriate - tempIDs[1] will either be old activeSemanticSceneID_ (if
-    // no semantic mesh was requested in loadStage); ID_UNDEFINED if desired
-    // was not found; activeSceneID_, or a unique value, the last of which means
-    // the semantic scene mesh is loaded.
-
-    if (activeSemanticSceneID_ != tempIDs[1]) {
-      // id has changed so act - if ID has not changed, do nothing
-      activeSemanticSceneID_ = tempIDs[1];
-      if ((activeSemanticSceneID_ != ID_UNDEFINED) &&
-          (activeSemanticSceneID_ != activeSceneID_)) {
-        sceneID_.push_back(activeSemanticSceneID_);
-      } else {  // activeSemanticSceneID_ = activeSceneID_;
-        // instance meshes and suncg houses contain their semantic annotations
-        // empty scene has none to worry about
-        if (!(stageType == assets::AssetType::SUNCG_SCENE ||
-              stageType == assets::AssetType::INSTANCE_MESH ||
-              stageFilename.compare(assets::EMPTY_SCENE) == 0)) {
-          // TODO: programmatic generation of semantic meshes when no
-          // annotations are provided.
-          LOG(WARNING) << ":\n---\n The active scene does not contain semantic "
-                          "annotations. \n---";
-        }
-      }
-    }  // if ID has changed - needs to be reset
-  }    // if (config_.createRenderer)
+  // TODO : reset may eventually have all the scene instance instantiation code
+  // so that scenes can be reset
+  reset();
+
+  return true;
+}  // Simulator::createSceneInstance
+
+bool Simulator::createSceneInstanceNoRenderer(
+    const std::string& activeSceneName) {
+  // Initial setup for scene instancing without renderer - sets or creates the
+  // current scene instance to correspond to the given name.  Also builds
+  // navmesh and semantic scene descriptor file if appropriate.
+  metadata::attributes::SceneAttributes::cptr curSceneInstanceAttributes =
+      setSceneInstanceAttributes(activeSceneName);
+
+  // TODO : reset may eventually have all the scene instance instantiation code
+  // so that scenes can be reset
+  reset();
+  return true;
+}  // Simulator::createSceneInstanceNoRenderer
+
+bool Simulator::createPathfinder(const std::string& navmeshFilename) {
```

Comment:
Just confused why this method was abstracted out. Shouldn't default navmesh recomputation be factored out of the constructor too?. This just seems like it does create + load. Maybe these logging should be moved to the pathfinder's loadNavMesh logic somewhere?

Reply:
Feel like this should live as a static method on the pathfinder class, don't you think? No simulator or pathfinder specific variables are used, feel like this should just be a constructor for Pathfinder that takes in a navmesh string. You can then use pathfinder.isloaded to figure out whether or not it loaded a navmesh.

Reply:
I like that.

Reply:
In an effort to minimize the impact this PR has, I am going to remove the mentioned methods and move the code contained to the only place the method was called.  The future design considerations this was laying the groundwork for are not yet on the horizon anyway.